components/trezor_bridge/trezor_bridge_handler.cc

```diff
@@ -0,0 +1,185 @@
+#include "net/http/http_request_headers.h"
+#include "net/http/http_response_headers.h"
+#include "net/http/http_status_code.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/resource/resource_bundle.h"
+#include "url/gurl.h"
+
+namespace {
+
+const base::flat_map<std::string, int> trezor_data_resources = {
+    {"./data/config.json", IDR_TREZOR_BRIDGE_IFRAME_DATA_CONFIG},
+    {"./data/coins.json", IDR_TREZOR_BRIDGE_IFRAME_DATA_COINS},
+    {"./data/messages/messages.json",
+     IDR_TREZOR_BRIDGE_IFRAME_DATA_MESSAGES_MESSAGES},
+    {"./data/messages/messages-v6.json",
+     IDR_TREZOR_BRIDGE_IFRAME_DATA_MESSAGES_MESSAGES_V6},
+    {"./data/messages/messages-v7.json",
+     IDR_TREZOR_BRIDGE_IFRAME_DATA_MESSAGES_MESSAGES_V7},
+    {"./data/messages/messages-v8.json",
+     IDR_TREZOR_BRIDGE_IFRAME_DATA_MESSAGES_MESSAGES_V8},
+    {"./data/firmware/1/releases.json",
+     IDR_TREZOR_BRIDGE_IFRAME_DATA_FIRMWARE_RELEASES1},
+    {"./data/firmware/2/releases.json",
+     IDR_TREZOR_BRIDGE_IFRAME_DATA_FIRMWARE_RELEASES2},
+    {"./data/bridge/releases.json",
+     IDR_TREZOR_BRIDGE_IFRAME_DATA_BRIDGE_RELEASES},
+};
+
+// TrezorSuite local address.
+const char kTrezorSuiteURL[] = "http://127.0.0.1:21325";
+
+net::NetworkTrafficAnnotationTag GetTrezorBridgeNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("trezor_bridge", R"(
+          semantics {
+            sender: "Trezor Bridge"
+            description:
+              "This bridge is used to communicate with TrezorSuite app "
+            trigger:
+              "Triggered by actions in brave://wallet."
+            data:
+              "Options of the commands."
+            destination: WEBSITE
+          }
+        )");
+}
+
+std::unique_ptr<network::SimpleURLLoader> CreateURLLoader(
+    const GURL& gurl,
+    const std::string& method,
+    const std::string& body) {
+  std::unique_ptr<network::ResourceRequest> request =
+      std::make_unique<network::ResourceRequest>();
+  request->url = gurl;
+  request->method = method;
+  // Use trezor.io origin because Trezor Sute accepts connections only from
+  // trezor.io domains.
+  request->headers.SetHeader(net::HttpRequestHeaders::kOrigin,
+                             "https://brave.trezor.io");
+  request->headers.SetHeader(net::HttpRequestHeaders::kContentType,
+                             "text/plain");
+
+  auto url_loader = network::SimpleURLLoader::Create(
+      std::move(request), GetTrezorBridgeNetworkTrafficAnnotationTag());
+  if (!body.empty())
+    url_loader->AttachStringForUpload(body, "text/plain");
+  return url_loader;
+}
+
+}  // namespace
+
+TrezorBridgeHandler::TrezorBridgeHandler(
```

Comment:
what is going on here? This is not the correct way to load resources from a pak file. cc @petemill